sql/codegen_xgboost.go

```diff
@@ -639,3 +655,84 @@ func xgFillDatabaseInfo(r *xgboostFiller, db *DB) (*xgboostFiller, error) {
 	}
 	return r, nil
 }
+
+func xgCreatePredictionTable(pr *extendedSelect, r *xgboostFiller, db *DB) error {
+	dropStmt := fmt.Sprintf("drop table if exists %s;", r.OutputTable)
+	if _, e := db.Exec(dropStmt); e != nil {
+		return fmt.Errorf("failed executing %s: %q", dropStmt, e)
+	}
+
+	fts, e := verify(pr, db)
+	if e != nil {
+		return e
+	}
+
+	var b bytes.Buffer
+	fmt.Fprintf(&b, "create table %s (", r.OutputTable)
+	for _, col := range r.AppendColumns {
+		typ, ok := fts.get(col)
+		if !ok {
+			return fmt.Errorf("xgCreatePredictionTable: Cannot find type of field %s", col)
+		}
+		stype, e := universalizeColumnType(db.driverName, typ)
+		if e != nil {
+			return e
+		}
+		fmt.Fprintf(&b, "%s %s, ", col, stype)
+	}
+	// add prob column
+	if len(r.ProbColumn) > 0 {
+		stype, e := universalizeColumnType(db.driverName, "DOUBLE")
+		if e != nil {
+			return e
+		}
+		fmt.Fprintf(&b, "%s %s, ", r.ProbColumn, stype)
+	}
+	// add detail column
+	if len(r.DetailColumn) > 0 {
+		stype, e := universalizeColumnType(db.driverName, "VARCHAR")
+		if e != nil {
+			return e
+		}
+		fmt.Fprintf(&b, "%s %s, ", r.DetailColumn, stype)
+	}
+	// add encoding column
+	if len(r.EncodingColumn) > 0 {
```

Comment:
What is the usage of the encoding column?

Reply:
Encoding column stores leaf indices of this sample in each tree. We transform leaf indices in  a string which format like: "index_0,index_1,......,index_n"